core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -73,6 +80,30 @@ public SnowflakeSqlDialect(Context context) {
     }
   }
 
+  @Override public SqlNode rewriteMaxMinExpr(SqlNode aggCall, RelDataType relDataType) {
+    return rewriteMaxMin(aggCall, relDataType);
+  }
+
+  /**
+   * Helper for rewrites of MAX/MIN.
+   * Snowflake, rewrite as
+   * BOOLOR_AGG/BOOLAND_AGG if the return type is BOOLEAN.
+   */
+  public static SqlNode rewriteMaxMin(SqlNode aggCall, RelDataType relDataType) {
```

Comment:
You should add your jira link or add description where you tested

Reply:
Consider adding your description, Add the specific SQL after the rewrite.